internal/wasm/module.go

```diff
@@ -169,10 +170,14 @@ type Module struct {
 	// IsHostModule true if this is the host module, false otherwise.
 	IsHostModule bool
 
-	// FunctionDefinitionSection is a wazero-specific section built on Validate.
+	functionDefinitionSectionWriteMutex sync.Mutex
+
+	FunctionDefinitionSectionInitialized uint32
```

Comment:
I would suggest to use `sync.Once` instead of combining these uint32 and mutext manually

Reply:
thanks!